bitcoin/src/amount.rs

```diff
@@ -2268,4 +2268,30 @@ mod tests {
             }
         }
     }
+
+    #[test]
+    #[cfg(feature = "serde")]
+    fn round_trip_flattened_option_signed_amount() {
+        #[derive(Serialize, Deserialize, Debug, PartialEq)]
+        #[serde(crate = "actual_serde")]
+        struct Foo {
+            #[serde(flatten)]
+            bar: Bar,
+        }
+
+        #[derive(Serialize, Deserialize, Debug, PartialEq)]
+        #[serde(crate = "actual_serde")]
+        struct Bar {
+            #[serde(with = "crate::amount::serde::as_btc::opt")]
+            amount: Option<SignedAmount>,
```

Comment:
If this PR ever gets rebased perhaps use `Amount` instead of `SignedAmount`, just because it saves devs wondering if this has anything to do with the signed-ness - which it doesn't since we use the same `opt` module for both.

Reply:
Ideally have both tests. I thought it has something to do with signedness and potentially `isize` vs `i64`. I suggested `i64` because that matches the inteded behavior of `SignedAmount` most closely.

I don't have much time for this now but my suggestion is to look into the implementation of `i64` and `u64` and compare the code.

Reply:
Our implementation of `visit_none` is not being called when the `flatten` attribute is present but it is called when the attribute is not present - that is why I think its a bug in serde. And to verify it should work I did the "replace our type with a type from the stdlib".